### PR TITLE
fix: command injection of CodeQL warning

### PR DESCRIPTION
**Pull request by lijie-lee**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/28823894
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/28823895
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/28823896

### CODE REVIEW

File: packages/fx-core/src/component/deps-checker/util/cpUtils.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import * as cp from "child_process";
 import * as os from "os";
 
+const shellQuote = require("shell-quote");
```
Comment (qinezh) [inline packages/fx-core/src/component/deps-checker/util/cpUtils.ts:8]: how about using the ESM syntax (`import xxx from yyy) for consistency?

**Review by qinezh — DISMISSED** [review]

File: packages/fx-core/src/component/deps-checker/util/cpUtils.ts (hunk 1 above)
Comment (lijie-lee) [reply packages/fx-core/src/component/deps-checker/util/cpUtils.ts:8]: Sure, Just found the type definition for "shell-quote". I'll have a quick update for that.

**Review by qinezh — APPROVED** [review]

File: packages/fx-core/src/component/local/localCertificateManager.ts [hunk 2]
```diff
@@ -381,14 +382,15 @@ export class LocalCertificateManager {
   }
 
   private async checkCertificateWindows(thumbprint: string): Promise<boolean> {
+    const quotedThumbprint: string = shellQuote.quote([thumbprint]);
     try {
       // try powershell first
-      const getCertCommand = `Get-ChildItem -Path Cert:\\CurrentUser\\Root | Where-Object { $_.Thumbprint -match '${thumbprint}' }`;
+      const getCertCommand = `Get-ChildItem -Path Cert:\\CurrentUser\\Root | Where-Object { $_.Thumbprint -match '${quotedThumbprint}' }`;
```
Comment (swatDong) [inline packages/fx-core/src/component/local/localCertificateManager.ts:388]: There's already single-quote in the command. Will this cause syntax error?

Comment (swatDong) [reply packages/fx-core/src/component/local/localCertificateManager.ts:388]: The thumbprint has a fixed format, why quote is needed?

**Review by a1exwang — APPROVED** [review]

File: packages/fx-core/src/component/local/localCertificateManager.ts (hunk 2 above)
Comment (lijie-lee) [reply packages/fx-core/src/component/local/localCertificateManager.ts:388]: Technically there is no need. To quote it just to meet the CodeQL checking.

**Review by wenytang-ms — DISMISSED** [review]

**Review by LongOddCode — DISMISSED** [review]

File: packages/fx-core/src/component/local/localCertificateManager.ts (hunk 2 above)
Comment (swatDong) [reply packages/fx-core/src/component/local/localCertificateManager.ts:388]: So would this change break the command?
E.g., if the `thumbprint` is `ABC123EF`, would the command be `... -match ''ABC123EF''`? Then the whole functionality is broken.

**Review by swatDong — APPROVED** [review]

**Review by a1exwang — APPROVED** [review]

**Review by wenytang-ms — APPROVED** [review]

**Pull request merged**